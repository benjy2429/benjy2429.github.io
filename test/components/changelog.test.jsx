/**
 * @jest-environment jsdom
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Changelog from '../../src/components/changelog';

const defaultProps = {
  items: [
    {
      version: 'v2',
      date: '2000-12-25',
      description: 'v2 description'
    },
    {
      version: 'v1',
      date: '1999-01-01',
      description: 'v1 description'
    }
  ]
};

describe('Changelog component', () => {
  beforeAll(() => {
    Date.now = jest.fn().mockReturnValue(new Date('2000-12-25'));
  });

  afterAll(() => {
    jest.resetAllMocks();
  });

  it('renders correctly without items', () => {
    const component = shallow(<Changelog />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly with items', () => {
    const component = shallow(<Changelog {...defaultProps} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders correctly without dates', () => {
    const component = shallow(
      <Changelog
        items={[
          {
            version: 'v1',
            date: '',
            description: 'test'
          }
        ]}
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
