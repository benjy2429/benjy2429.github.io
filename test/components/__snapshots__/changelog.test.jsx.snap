// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Changelog component renders correctly with items 1`] = `
<div>
  <h1>
    Changelog
  </h1>
  <div
    className="changelog-item"
    key="v2"
  >
    <p>
      <span
        className="changelog-version"
      >
        v2
      </span>
      <br />
      <span
        className="changelog-date"
        title="25th Dec 2000"
      >
        a few seconds ago
      </span>
    </p>
    <p
      className="changelog-description"
    >
      v2 description
    </p>
  </div>
  <div
    className="changelog-item"
    key="v1"
  >
    <p>
      <span
        className="changelog-version"
      >
        v1
      </span>
      <br />
      <span
        className="changelog-date"
        title="1st Jan 1999"
      >
        2 years ago
      </span>
    </p>
    <p
      className="changelog-description"
    >
      v1 description
    </p>
  </div>
</div>
`;

exports[`Changelog component renders correctly without dates 1`] = `
<div>
  <h1>
    Changelog
  </h1>
  <div
    className="changelog-item"
    key="v1"
  >
    <p>
      <span
        className="changelog-version"
      >
        v1
      </span>
    </p>
    <p
      className="changelog-description"
    >
      test
    </p>
  </div>
</div>
`;

exports[`Changelog component renders correctly without items 1`] = `
<div>
  <h1>
    Changelog
  </h1>
</div>
`;
